Source/SvgElement.cs

```diff
@@ -312,7 +312,7 @@ public SvgCustomAttributeCollection CustomAttributes
             get { return this._customAttributes; }
         }
 
-        private readonly Matrix _zeroMatrix = new Matrix(0, 0, 0, 0, 0, 0);
+        private static readonly Matrix _zeroMatrix = new Matrix(0, 0, 0, 0, 0, 0);

```

Comment:
I wonder how this will behave on Mac/Unix without the GDI+ package. There is a test to validate the compatibility which generates a Matrix to test if the GDI+ package works (and will yield a specific exception which is caught and rewrapped in a more sane error), if the _zeroMatrix is now a static property I think this will be initialized somewhere in the beginning, triggering the GDI+ exception on a not really expected location. Not really sure when it will be initialized, but I guess this might break Linux and Mac compatibility.

Also, why make this static? If this element is nicely disposed upon disposal of the SvgElement it should be nicely disposed when the SvgElement is being disposed.

Reply:
@gvheertum 

This is because `Matrix` is generated for each instance of `SvgElement` unless it is made `static`.

Reply:
Fair point, still wondering what happens on systems having the GDI+ problem (Mac and Linux without the proper packages). Maybe we should wrap it in a static property that initializes the matrix on the first request?

`
private static Matrix _zeroMatrix;
private static Matrix ZeroMatrix
{
    get { return _zeroMatrix = _zeroMatrix ?? new Matrix(0, 0, 0, 0, 0, 0);
}
`
And then use the property instead of the field. You can consider try doing this in the constructor (parallelization might get interesting there if multiple instances try to set/check the property, so we might need locking here). My biggest fear is that if the Matrix construction is called on a unexpected location (I haven't got the faintest idea where it will be constructed if it's a static field) the library might have some weird behavior.

Doing the construction on an expected location (in the property or the constructor) might prevent issues on Mac/Linux, because afaik the error occurs there on the constructor of the Matrix. There is some code handling the GDI+ errors and I'm afraid this might break now.